crates/tako/src/gateway.rs

```diff
@@ -105,6 +105,36 @@ pub struct StopWorkerRequest {
     pub worker_id: WorkerId,
 }
 
+#[derive(Serialize, Deserialize, Debug)]
+pub struct WorkerTypeQuery {
+    pub descriptor: ResourceDescriptor,
+    pub max_sn_workers: u32,            // For single-node tasks
+    pub max_worker_per_allocation: u32, // For multi-node tasks
+}
+
+/* Ask scheduler for the information about how
```

Comment:
Could you please copy this comment or some shortened version also to the `compute_new_worker_query` function? I was quite lost until I got here during review.